pallets/gear/src/queue.rs

```diff
@@ -17,6 +17,149 @@
+
+        // The second step is to load instrumented binary code of the program but
+        // first its correct length should be obtained.
+        let context = match core_processor::precharge_for_code_length(
+            block_config,
+            precharged_dispatch,
+            program_id,
+            actor_data,
+        ) {
+            Ok(context) => context,
+            Err(journal) => return Ok(journal),
+        };
+
+        // Load correct code length value.
+        let code_id = context.actor_data().code_id;
+        let code_len_bytes = T::CodeStorage::get_code_len(code_id).unwrap_or_else(|| {
+            unreachable!(
+                "Program '{:?}' exists so do code len '{:?}'",
+                program_id, code_id
+            )
+        });
+
+        // Adjust gas counters for fetching instrumented binary code.
+        let context =
+            match core_processor::precharge_for_code(block_config, context, code_len_bytes) {
+                Ok(context) => context,
+                Err(journal) => return Ok(journal),
+            };
+
+        // Load instrumented binary code from storage.
+        let code = T::CodeStorage::get_code(code_id).unwrap_or_else(|| {
+            unreachable!(
+                "Program '{:?}' exists so do code '{:?}'",
+                program_id, code_id
+            )
+        });
+
+        // Reinstrument the code if necessary.
+        let schedule = T::Schedule::get();
+        let (code, context) =
+            if code.instruction_weights_version() == schedule.instruction_weights.version {
+                (code, ContextChargedForInstrumentation::from(context))
+            } else {
+                log::debug!("Re-instrumenting code for program '{:?}'", program_id);
+
+                let context = match core_processor::precharge_for_instrumentation(
+                    block_config,
+                    context,
+                    code.original_code_len(),
+                ) {
+                    Ok(context) => context,
+                    Err(journal) => return Ok(journal),
+                };
+
+                (Pallet::<T>::reinstrument_code(code_id, &schedule), context)
+            };
+
+        // The last one thing is to load program memory. Adjust gas counters for memory pages.
+        let context = match core_processor::precharge_for_memory(block_config, context) {
+            Ok(context) => context,
+            Err(journal) => return Ok(journal),
+        };
+
+        // Load program memory pages.
+        let memory_pages = Pallet::<T>::get_and_track_memory_pages(
+            ext_manager,
+            program_id,
+            &context.actor_data().pages_with_data,
+            lazy_pages_enabled,
+        )
+        .ok_or(QueueStepError::NoMemoryPages)?;
+
+        let (random, bn) = T::Randomness::random(dispatch_id.as_ref());
+        let origin = ProgramId::from_origin(external.into_origin());
```

Comment:
gr_origin is removed, so do we need this though?

Reply:
`gr_origin` is a user space call but on the backend side we could operate with the message origin

Reply:
do we really? I see not usage of `origin` field in `ProcessorContext`, except for `gr_origin`

Reply:
> `gr_origin` is a user space call but on the backend side we may operate with the message origin

Totally unused and shouldn't be used further